File: 프로그래머스/unrated/181926. 수 조작하기 1/수 조작하기 1.js
function solution(n, control) {
    // 방법 2 - reduce 함수로 각 알파벳에 접근하고, switch 함수로 각 알파벳에 맞는 연산을 한다.
    return [...control].reduce((acc, el) => {
        switch(el) {
            case 'w':
                return acc += 1
            case 's':
                return acc -= 1
            case 'd':
                return acc += 10
            case 'a':
                return acc -= 10
        }
    }, n)
    
//     // 방법 1
//     // 객체의 키를 알파벳으로, 값을 n에 더할 값을 설정
//     const obj = {
//         'w': 1,
//         's': -1,
//         'd': 10,
//         'a': -10
//     }
    
//     // reduce 함수로 각 알파벳에 해당하는 값을 n에 더한다.
//     return [...control].reduce((acc, el) => acc += obj[el], n)
}

File: 프로그래머스/unrated/181926. 수 조작하기 1/수 조작하기 1.py
def solution(n, control):
    # 딕셔너리 사용 - 알파벳을 키로, n에 더해줄 값을 값으로 설정
    dic = {'w': 1, 's': -1, 'd': 10, 'a': -10}
    
    for ch in control:
        n += dic[ch]
    
    return n